Objective-C/CBLMutableDictionary.mm

```diff
@@ -143,6 +153,39 @@ - (void) setData: (nullable NSDictionary<NSString*,id>*)data {
     }
 }
 
+- (BOOL) setJSON: (NSString*)json error: (NSError**)error {
+    CBLStringBytes jsonSlice(json);
+    FLError flEerror = {};
+    Encoder enc;
+    if (!FLEncoder_ConvertJSON(enc, jsonSlice)) {
+        flEerror = enc.error();
+        CBLWarnError(Database, @"%@: Error converting JSON %d", self, flEerror);
+        convertError(flEerror, error);
+        return NO;
+    }
+    
+    flEerror = {};
+    FLSliceResult result = FLEncoder_Finish(enc, &flEerror);
+    if (!result.buf) {
+        CBLWarnError(Database, @"%@: Error decoding JSON: %d", self, flEerror);
+        convertError(flEerror, error);
+        return NO;
+    }
+    
+    FLDict dict = FLValue_AsDict(FLValue_FromData(C4Slice(result), kFLTrusted));
+    _dict.clear();
```

Comment:
Can we just replace _array with a new MDict? There is `MDict(MValue *mv, MCollection *parent)` that can be used to create a new Array with `dict` and the parent from `_array.parent`.

Reply:
https://issues.couchbase.com/browse/CBL-1760